dts/msm8916/msm8916-512mb-mtp.dts

```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	compatible = "qcom,msm8916-512mb-mtp", "qcom,msm8916";
+	qcom,msm-id = <206 0>;
+	qcom,board-id = <8 0x100>;
+	
+	zhihe-ufi-001c {
+		model = "ufi-001b/ufi-001c 4G Modem Stick";
+		compatible = "zhihe,ufi-001c", "qcom,msm8916", "lk2nd,device";
+
+		/* Register EDL button as HOME button */
+		/* This is the only button on this device */
+		lk2nd,keys = <KEY_HOME 37 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
```

Comment:
Does the button work correctly?
That is, assuming you don't have anything to boot into, the lk log shall have an error like:
```
[340] ERROR: Could not do normal boot. Reverting to fastboot mode.
```
which then should not be in the log if the device is booted with the button pressed.

You can get the log with `fastboot oem lk_log && fastboot get_staged /dev/stdout` or from the uart

I don't think I saw active-high buttons that often, and the board I have (an "uf896" to which I never really got my hands for a while), has it's key as active-low a usual...

Reply:
> Does the button work correctly? That is, assuming you don't have anything to boot into, the lk log shall have an error like:
> 
> ```
> [340] ERROR: Could not do normal boot. Reverting to fastboot mode.
> ```
> 
> which then should not be in the log if the device is booted with the button pressed.
> 
> You can get the log with `fastboot oem lk_log && fastboot get_staged /dev/stdout` or from the uart
> 
> I don't think I saw active-high buttons that often, and the board I have (an "uf896" to which I never really got my hands for a while), has it's key as active-low a usual...

Yes, at least lk1st successfully enter fastboot mode by pressing that button with current dtb.
compile command: `make TOOLCHAIN_PREFIX=xxx LK1ST_DTB=msm8916-512mb-mtp lk1st-msm8916`